core/http/src/main/java/org/trellisldp/http/impl/PostHandler.java

```diff
@@ -77,108 +86,153 @@
+                mayContinue(false);
+                return completedFuture(err);
             }
-            final IRI container = rdf.createIRI(TRELLIS_DATA_PREFIX + req.getPath());
-            final Future<Boolean> success = trellis.getResourceService().create(internalId, session, ldpType,
-                    dataset.asDataset(), container, binary);
-            if (success.get()) {
-
-                // Add Audit quads
-                try (final TrellisDataset auditDataset = TrellisDataset.createDataset()) {
-                    trellis.getAuditService().creation(internalId, session).stream()
-                        .map(skolemizeQuads(trellis.getResourceService(), baseUrl)).forEachOrdered(auditDataset::add);
-                    if (!trellis.getResourceService().add(internalId, session, auditDataset.asDataset()).get()) {
-                        LOGGER.error("Using AuditService {}", trellis.getAuditService());
-                        LOGGER.error("Unable to act against resource at {}", internalId);
-                        LOGGER.error("because unable to write audit quads: \n{}",
-                                        auditDataset.asDataset().stream().map(Quad::toString).collect(joining("\n")));
-                        throw new BadRequestException("Unable to write audit information. Please consult "
-                                + "the logs for more information.");
-                        }
-                }
-
-                // Add memento data
-                trellis.getResourceService().get(internalId).ifPresent(trellis.getMementoService()::put);
-
-                final ResponseBuilder builder = status(CREATED).location(create(identifier));
-
-                // Add LDP types
-                ldpResourceTypes(ldpType).map(IRI::getIRIString).forEach(type -> builder.link(type, "type"));
-
-                return builder;
+
+            // Check for any constraints
+            final ResponseBuilder constraints = checkConstraint(mutable.getGraph(PreferUserManaged).orElse(null),
+                    ldpType, ofNullable(rdfSyntax).orElse(TURTLE));
+            if (nonNull(constraints)) {
+                mayContinue(false);
+                return completedFuture(constraints);
             }
-            throw new BadRequestException("Unable to save resource to persistence layer. Please consult the logs "
-                    + "for more information.");
 
-        } catch (final InterruptedException | ExecutionException ex) {
-            LOGGER.error("Error persisting data", ex);
+            binary = null;
         }
 
-        LOGGER.error("Unable to persist data to location at {}", internalId.getIRIString());
-        return serverError().type(TEXT_PLAIN_TYPE)
-            .entity("Unable to persist data. Please consult the logs for more information");
+        // Should this come from the parent resource data?
+        final IRI container = rdf.createIRI(TRELLIS_DATA_PREFIX + getRequest().getPath());
```

Comment:
Is this assuming something about the relationship of the URIs for container and contained?

Reply:
Good call. This data can be set when the `PostHandler` is initially constructed.